### ISSUE TITLE
ONNX エクスポーターの verify が出力を位置インデックスで取得している

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochidetection#99]
## Summary

`OnnxExporter.verify()` が ONNX 出力を位置インデックス (`onnx_outputs[0]`, `[1]`) で取得している. エクスポート時に `["logits", "pred_boxes"]` と名前を指定しているが, ONNX 最適化パスで出力順が変わると無言で不正な比較になる. `OnnxBackend` は名前ベースで解決しており不整合.

## Steps to Reproduce

1. ONNX モデルをエクスポートする.
2. ONNX 最適化ツール (onnxsim 等) で最適化し, 出力順が変わる.
3. `verify()` が logits と pred_boxes を逆に比較し, 検証が不正に失敗する.

## Expected Behavior

出力を名前ベースで取得し, `OnnxBackend` と整合的な方法で比較する.

## Actual Behavior

`onnx/exporter.py` L159-160 で位置インデックスを使用している.

```python
# onnx/exporter.py L159-160
onnx_logits = onnx_outputs[0]
onnx_boxes = onnx_outputs[1]
```

## Labels

`bug`

## Branch

`fix/onnx-verify-output-by-name`
